### PR TITLE
feat(BMS): bms instance support edit nics

### PR DESCRIPTION
**Pull request by MrLiangSir**
**What this PR does / why we need it**:
bms instance support edit nics

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:
1.Security group parammeters are missing in the network card, this is a legacy feature.
2.The main network card is a network card configured with routing rules on the BMS server and cannot be unbound.

**Release note**:

```release-note
1. Support edit nics attribute information when updating a BMS instance
2. Remove the relevant forceNew attribute in nics.
3. Modify the forceNew attribute in nics in the document.
4. Add test cases related to editing nics.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/bms TESTARGS='-run TestAccBmsInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/bms -v -run TestAccBmsInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccBmsInstance_basic
=== PAUSE TestAccBmsInstance_basic
=== CONT  TestAccBmsInstance_basic
--- PASS: TestAccBmsInstance_basic (607.27s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/bms       607.314s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
- please submit the reconstruction code related to the EIP and BMS test modifying to a new commit.
- please update the PR description, lots of comments are missing, title ditto.

**Comment by MrLiangSir** [comment]
make testacc TEST=./huaweicloud/services/acceptance/bms TESTARGS='-run TestAccBmsInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/bms -v -run TestAccBmsInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccBmsInstance_basic
=== PAUSE TestAccBmsInstance_basic
=== CONT  TestAccBmsInstance_basic
--- PASS: TestAccBmsInstance_basic (576.44s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/bms       576.482s

File: huaweicloud/services/acceptance/bms/resource_huaweicloud_bms_instance_test.go [hunk 1]
```diff
@@ -179,7 +181,7 @@ resource "huaweicloud_vpc_eip" "myeip" {
   publicip {
     type = "5_bgp"
   }
-
+
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/bms/resource_huaweicloud_bms_instance_test.go]: This line has two extra spaces

File: docs/resources/bms_instance.md [hunk 2]
```diff
@@ -187,11 +187,9 @@ installed, the `admin_pass` field becomes invalid.
 
 The `nics` block supports:
 
-* `subnet_id` - (Required, String, ForceNew) Specifies the ID of subnet to attach to the instance. Changing this creates
-  a new instance.
+* `subnet_id` - (Required, String) Specifies the ID of subnet to attach to the instance.
 
-* `ip_address` - (Optional, String, ForceNew) Specifies a fixed IPv4 address to be used on this network. Changing this
-  creates a new instance.
+* `ip_address` - (Optional, String) Specifies a fixed IPv4 address to be used on this network.
```
Comment (deer-hang) [inline docs/resources/bms_instance.md:199]: Please check this parameter in test case.

**Comment by MrLiangSir** [comment]
make testacc TEST=./huaweicloud/services/acceptance/bms TESTARGS='-run TestAccBmsInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/bms -v -run TestAccBmsInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccBmsInstance_basic
=== PAUSE TestAccBmsInstance_basic
=== CONT  TestAccBmsInstance_basic
--- PASS: TestAccBmsInstance_basic (685.23s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/bms       685.291s

**Comment by MrLiangSir** [comment]
make testacc TEST=./huaweicloud/services/acceptance/bms TESTARGS='-run TestAccBmsInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/bms -v -run TestAccBmsInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccBmsInstance_basic
=== PAUSE TestAccBmsInstance_basic
=== CONT  TestAccBmsInstance_basic
--- PASS: TestAccBmsInstance_basic (768.31s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/bms       768.438s

**Comment by MrLiangSir** [comment]
make testacc TEST=./huaweicloud/services/acceptance/bms TESTARGS='-run TestAccBmsInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/bms -v -run TestAccBmsInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccBmsInstance_basic
=== PAUSE TestAccBmsInstance_basic
=== CONT  TestAccBmsInstance_basic
--- PASS: TestAccBmsInstance_basic (653.89s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/bms       653.941s

**Comment by MrLiangSir** [comment]
make testacc TEST=./huaweicloud/services/acceptance/bms TESTARGS='-run TestAccBmsInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/bms -v -run TestAccBmsInstance_basic -timeout 360m -parallel 4
=== RUN   TestAccBmsInstance_basic
=== PAUSE TestAccBmsInstance_basic
=== CONT  TestAccBmsInstance_basic
--- PASS: TestAccBmsInstance_basic (636.92s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/bms       637.024s

File: huaweicloud/services/bms/resource_huaweicloud_bms_instance.go [hunk 3]
```diff
@@ -66,19 +67,17 @@ func ResourceBmsInstance() *schema.Resource {
 			"nics": {
 				Type:     schema.TypeList,
 				Required: true,
-				ForceNew: true,
 				MaxItems: 2,
+				MinItems: 1,
```
Comment (ShiChangkuo) [inline huaweicloud/services/bms/resource_huaweicloud_bms_instance.go]: this line is no sense for a required list.
```suggestion
```

**Pull request merged**